acceptance-tests/mssql_data_migration_test.go

```diff
@@ -0,0 +1,175 @@
+
+		Expect(password).NotTo(BeEmpty())
+
+		By("creating a DMS source endpoint")
+		sourceCreds := legacyBinding.Credential()
+		var sourceReceiver struct {
+			DBName   string `mapstructure:"database"`
+			Server   string `mapstructure:"hostname"`
+			Username string `mapstructure:"username"`
+			Password string `mapstructure:"password"`
+			Port     int    `mapstructure:"port"`
+		}
+		Expect(mapstructure.Decode(sourceCreds, &sourceReceiver)).NotTo(HaveOccurred())
+		sourceEndpoint := dms.CreateEndpoint(dms.CreateEndpointParams{
+			EndpointType:    dms.Source,
+			EnvironmentName: metadata.Name,
+			Username:        masterUsername,
+			Password:        password,
+			Server:          sourceReceiver.Server,
+			DatabaseName:    sourceReceiver.DBName,
+			Region:          metadata.Region,
+			Engine:          "sqlserver",
+			Port:            sourceReceiver.Port,
+		})
+		defer sourceEndpoint.Cleanup()
+
+		By("creating a target service instance using the CSB")
+		csbServiceInstance := services.CreateInstance(
+			"csb-aws-mssql",
+			services.WithPlan("default"),
+			services.WithParameters(
+				map[string]any{
+					"multi_az":                false,
+					"backup_retention_period": 0,
+					"require_ssl":             false,
+				},
+			),
+		)
+		defer csbServiceInstance.Delete()
+
+		By("creating a DMS target endpoint")
+		csbKey := csbServiceInstance.CreateServiceKey()
+		defer csbKey.Delete()
+		var targetReceiver struct {
+			Username string `json:"username"`
+			Password string `json:"password"`
+			Server   string `json:"hostname"`
+			DBName   string `json:"name"`
+			Port     int    `json:"port"`
+		}
+		csbKey.Get(&targetReceiver)
+		targetEndpoint := dms.CreateEndpoint(dms.CreateEndpointParams{
+			EndpointType:    dms.Target,
+			EnvironmentName: metadata.Name,
+			Username:        targetReceiver.Username,
+			Password:        targetReceiver.Password,
+			Server:          targetReceiver.Server,
+			DatabaseName:    targetReceiver.DBName,
+			Region:          metadata.Region,
+			Engine:          "sqlserver",
+			Port:            targetReceiver.Port,
+		})
+		defer targetEndpoint.Cleanup()
+
+		By("waiting for the replication instance to be ready")
+		replicationInstance.Wait()
+
+		By("running the replication task")
+		dms.RunReplicationTask(replicationInstance, sourceEndpoint, targetEndpoint, metadata.Region, schema)
+
+		By("switching the app data source from the legacy MSSQL to the new CSB created MSSQL")
+		legacyBinding.Unbind()
+		csbServiceInstance.Bind(golangAppOne)
+		apps.Restart(golangAppOne)
+
+		By("checking that the data is available in the target CSB created database")
+
+		Expect(golangAppOne.GET("%s/%s?tls=disable", schema, key).String()).To(Equal(value))
+	})
+})
+
+func getPassword(cmd string, args ...string) string {
+	command := exec.Command(cmd, args...)
+	session, err := gexec.Start(command, GinkgoWriter, GinkgoWriter)
+	Expect(err).NotTo(HaveOccurred())
+	Eventually(session, time.Minute).Should(gexec.Exit(0))
```

Comment:
do we want to register this as a helper?

Reply:
great point! Yes. I will modify it now